tests/RobotFramework/tests/cumulocity/telemetry/child_device_telemetry.robot

```diff
@@ -79,6 +79,70 @@ Child device supports sending custom child device measurements directly to c8y
+    Cumulocity.Should Have Services    name=app2    min_count=1    max_count=1
+    
+    Execute Command    tedge mqtt pub te/device/${CHILD_SN}/service/app2/m/service_type002 '{"temperature": 30.1}'
+    Cumulocity.Device Should Exist    ${DEVICE_SN}:device:${CHILD_SN}:service:app2
+    ${measurements}=    Device Should Have Measurements    minimum=1    maximum=1    type=service_type002
+    Should Be Equal    ${measurements[0]["type"]}    service_type002
+    Should Be Equal As Numbers    ${measurements[0]["temperature"]["temperature"]["value"]}    30.1
+
+# alarms
+Send alarms to an unregistered child service
+    Execute Command    tedge mqtt pub te/device/${CHILD_SN}/service/app3/a/alarm_001 '{"text": "test alarm","severity":"major"}'
+    Cumulocity.Device Should Exist    ${CHILD_SN}
+    Cumulocity.Should Have Services    min_count=1    max_count=1    name=app3
+
+    Cumulocity.Device Should Exist    ${DEVICE_SN}:device:${CHILD_SN}:service:app3
+    ${alarms}=    Device Should Have Alarm/s    expected_text=test alarm    type=alarm_001    minimum=1    maximum=1
+    Should Be Equal    ${alarms[0]["type"]}    alarm_001
+    Should Be Equal    ${alarms[0]["severity"]}    MAJOR
+
+Send alarms to a registered child service
+    Execute Command    tedge mqtt pub --retain te/device/${CHILD_SN}/service/app4 '{"@type":"service","@parent":"device/${CHILD_SN}//"}'
+    Cumulocity.Device Should Exist    ${CHILD_SN}
+    Cumulocity.Should Have Services    name=app4    min_count=1    max_count=1
+
+    Execute Command    tedge mqtt pub te/device/${CHILD_SN}/service/app4/a/alarm_002 '{"text": "test alarm"}'
+    Cumulocity.Device Should Exist    ${DEVICE_SN}:device:${CHILD_SN}:service:app4
+    ${alarms}=    Device Should Have Alarm/s    expected_text=test alarm    type=alarm_002    minimum=1    maximum=1
+    Should Be Equal    ${alarms[0]["type"]}    alarm_002
+
+# events
+Send events to an unregistered child service
+    Execute Command    tedge mqtt pub te/device/${CHILD_SN}/service/app5/e/event_001 '{"text": "test event"}'
+    Cumulocity.Device Should Exist    ${CHILD_SN}
+    Cumulocity.Should Have Services    name=app5    min_count=1    max_count=1
+
+    Cumulocity.Device Should Exist    ${DEVICE_SN}:device:${CHILD_SN}:service:app5
+    Device Should Have Event/s    expected_text=test event    type=event_001    minimum=1    maximum=1
+
+Send events to a registered child service
+    Execute Command    tedge mqtt pub --retain te/device/${CHILD_SN}/service/app6 '{"@type":"service","@parent":"device/${CHILD_SN}//"}'
+    Cumulocity.Device Should Exist    ${CHILD_SN}
+    Cumulocity.Should Have Services    name=app6    min_count=1    max_count=1
+
+    Cumulocity.Device Should Exist    ${DEVICE_SN}:device:${CHILD_SN}:service:app6
+    Execute Command    tedge mqtt pub te/device/${CHILD_SN}/service/app6/e/event_002 '{"text": "test event"}'
+    Device Should Have Event/s    expected_text=test event    type=event_002    minimum=1    maximum=1
+
```

Comment:
You may want to add similar tests for services on nested child devices as well. The "unregistered nested child service" case is not even needed as nested child devices must have explicit registration.

Reply:
I will do it as part of another PR as there are unit tests are also missing.

Reply:
No we should include the tests here as well so that we are sure the code supports it before we merge it.

Reply:
Since that is not in the scope of this task, I do not want to bloat this task. The nested child devices are a completely different thing, I would do it as part of another PR because this needs Unit tests and integration tests.

One question: can nested child devices also have the services? they need telemetry support as well right?

Reply:
No I would disagree, it is very much in the task as a child device also reference to any child device regardless if it is an immediate child or a nested child.

And generally we should get in the habit of ensuring the code does what it is meant to before it gets merged.

Reply:
> One question: can nested child devices also have the services? they need telemetry support as well right?

Yes, services are supported on any type of child device (immediate or nested)

Reply:
ok, thanks for the clarification. I will add the tests.

Reply:
ok, thanks for the clarification. I will update the tests.